### PR TITLE
Support non-const scale arguments in rounding functions

### LINKED ISSUE #50420: Function `round()` with non-const second arguments
```
SELECT round(materialize(1 / 3), materialize(3))

Query id: 8f1ccbc9-d25d-42ed-93f2-16ad73c0e35f


0 rows in set. Elapsed: 0.762 sec. 

Received exception:
Code: 44. DB::Exception: Argument at index 1 for function round must be constant. (ILLEGAL_COLUMN)
```

### PR DESCRIPTION
**Pull request by jh0x**
Closes #50420

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Functions `round()`, `roundBankers()`, `floor()`, `ceil()` and `trunc()` now accept non-const scale arguments

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 16b198e75f0c3d47925b9765f6d4a46f3c233364 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55990/16b198e75f0c3d47925b9765f6d4a46f3c233364/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>❌ failure</td></tr>
<tbody>
</table>

**Comment by jh0x** [comment]
I'll have a look at this and fix it.

File: src/Functions/FunctionsRound.h [hunk 1]
```diff
@@ -494,6 +537,53 @@ struct Dispatcher
 
         return col_res;
     }
+
+    template<typename S>
```
[the remark this answers is not in the document]
Comment (jh0x) [reply src/Functions/FunctionsRound.h:541]: I will have a look.

**Comment by rschu1ze** [comment]
Resolved by https://github.com/ClickHouse/ClickHouse/pull/64798

**Pull request closed without merging**